consensus/phases/firstphase.go

```diff
@@ -169,6 +170,39 @@ func detectSparseBitsetLength(claims map[core.RecordRef][]packets.ReferendumClai
 	return 0, errors.New("not implemented")
 }
 
+func (fp *FirstPhase) getSignedClaims(claims []packets.ReferendumClaim) []packets.ReferendumClaim {
+	result := make([]packets.ReferendumClaim, 0)
+	for _, claim := range claims {
+		joinClaim, ok := claim.(*packets.NodeJoinClaim)
+		if ok {
+			signConfirmed, err := fp.claimSignIsOk(joinClaim)
+			if err != nil {
+				log.Error("[ getSignedClaims ] failed to check a claim sign")
+				continue
+			}
+			if !signConfirmed {
+				log.Error("[ getSginedClaims ] sign is unconfirmed")
+				continue
+			}
+		}
+		result = append(result, claim)
+	}
+	return result
+}
+
+func (fp *FirstPhase) claimSignIsOk(claim *packets.NodeJoinClaim) (bool, error) {
```

Comment:
move it to a separate function and cover with tests

Reply:
tests for sign/check sign will be available after certificate logic refactoring